server/src/main/java/org/elasticsearch/search/aggregations/bucket/geogrid/InternalGeoGrid.java

```diff
@@ -117,6 +118,20 @@ public InternalGeoGrid doReduce(List<InternalAggregation> aggregations, ReduceCo
         return create(getName(), requiredSize, Arrays.asList(list), pipelineAggregators(), getMetaData());
     }
 
+    public B reduceBucket(List<B> buckets, InternalAggregation.ReduceContext context) {
+        assert buckets.size() > 0;
+        List<InternalAggregations> aggregationsList = new ArrayList<>(buckets.size());
+        long docCount = 0;
+        for (B bucket : buckets) {
+            docCount += bucket.docCount;
+            aggregationsList.add(bucket.aggregations);
+        }
+        final InternalAggregations aggs = InternalAggregations.reduce(aggregationsList, context);
+        return createBucket(buckets.get(0), docCount, aggs);
+    }
+
+    abstract B createBucket(InternalGeoGridBucket bucket, long docCount, InternalAggregations aggregations);
```

Comment:
should we be consistent with other `createBucket`s and pass in `long hashAsLong` instead of `InternalGeoGridBucket bucket`?

Reply:
Ah good catch.  I had it that way originally, but `OrdinalBucket` messed it up.  With OrdinalBucket gone it's not a problem, ++